src/lib.rs

```diff
@@ -51,7 +49,7 @@ impl NS2Stats {
             if game.round_info.round_length < 300.0 {
                 continue;
             }
-            let map_entry = maps.entry(game.round_info.map_name).or_insert(Map {
+            let map_entry = maps.entry(game.round_info.map_name.clone()).or_insert(Map {
```

Comment:
I thought about that too, but I don't think `Cow` helps here, as we never modify the keys anyway. Converting a `String` to `&'static str` doesn't work either. However, I think we could avoid cloning if `NS2Stats` gets a lifetime parameter, so that the signature would be `pub fn compute(games: &'a [GameStats]) -> NS2Stats<'a>`.

Reply:
A different solution may be to use `Rc<str>` instead of `String` (already when reading the JSON), although I'm not sure it's possible to deserialize a string into an `Rc<str>`.

Reply:
Serde can deserialize a string into `Rc<str>` with the feature `rc`
https://serde.rs/feature-flags.html#-features-rc

Reply:
I ended up using `&'a str`.